### DISCUSSION TITLE
[Unknown Mod][Game] Allow to dismiss a hero when he is located in a castle

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#4758] [Ideas]
Allow to dismiss a hero when he is located in a castle.
I see no reason why this should not be possible.

In fheroes2:

The hero is located in a castle.
![Dismiss hero in castle 001](https://user-images.githubusercontent.com/43583994/144881615-2c511774-fc4a-4868-acd0-1f9eda16ecef.png)

It is not possible to dismiss the hero.
![Dismiss hero in castle 002](https://user-images.githubusercontent.com/43583994/144881644-399d99b5-5967-40ec-95bc-03fcfd290e69.png)

After it will be possible to dismiss a hero located in a castle, the game simply displays the updated castle screen (with a fade to black as in the original game when it will be implemented).
![Dismiss hero in castle 003](https://user-images.githubusercontent.com/43583994/144881656-f0620c75-a7bc-4dbb-8fc1-827dc681c4d4.png)

Or the game simply displays the updated Kingdom Overview Towns/Castles screen if it was the starting screen.
![Dismiss hero in castle 005a](https://user-images.githubusercontent.com/43583994/144884126-c9e303ea-2463-4bc8-893e-d7a491069db4.png)

Or the game simply displays the updated Kingdom Overview Heroes screen if it was the starting screen.
![Dismiss hero in castle 006](https://user-images.githubusercontent.com/43583994/144884143-be58817f-58aa-4103-ad38-cb8cd9155159.png)

This is not an option.
This can be implemented directly in fheroes2.

**Comment by Branikolog** [comment]
Hello, @LeHerosInconnu.
I also thought about that, but could this happen, that inability to dismiss a hero in a castle is a gameplay feature of Heroes 2? If I'm not mistaken in vanilla H3 you cannot dismis hero in a castle too. Should we take away all known gameplay behaviour?

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Hello, @LeHerosInconnu. I also thought about that, but could this happen, that inability to dismiss a hero in a castle is a gameplay feature of Heroes 2? If I'm not mistaken in vanilla H3 you cannot dismis hero in a castle too. Should we take away all known gameplay behaviour?

Let's consider a case in example.
A hero of the player is visiting one of his castles.
The player looks at the screens of the heroes available in the castle and wants to recruit one of these two heroes.
Back at the castle screen, the player has to exit the castle screen, take his hero out of the castle, open the hero screen and dismiss him, and then reopen the castle screen (which may not be selected if the player has several heroes and may also be difficult to find if the player has many castles).

Now, why should the player do all these actions, when it would be much more convenient if once back at the castle screen he opens the hero screen and dismisses him directly.

Is there any advantage to keep the original way of working?

**Reply by Branikolog** [reply]
Could it be a restriction rules of the OG? Like 8 hero limit.

**Reply by LeHerosInconnu** [reply]
Hello @Branikolog,

> Could it be a restriction rules of the OG? Like 8 hero limit.

The only eventual restriction rule I can see is that if an opposing hero (or an allied hero who has no more movement points and that the player wants to keep) is right in front of the castle entrance, the hero who is visiting the castle cannot leave and cannot be dismissed.
The same applies if the visiting hero has no more movement points.
But to me this is more of a side effect than a clearly thought out and explicitly created restriction rule.

The case of the opposing hero who blocks the exit (only this case) can be kept as an option for the purists. :)
But this happens very rarely, especially since in the original game, AI heroes can attack a city "obliquely" without being precisely on the case just below the active castle square (certainly a special bonus for the AI... or a bug). :)
So yet another option for purists? :D

**Reply by Branikolog** [reply]
I'm really hesitating about that. On one hand, allowing this makes gameplay more convenient. 
But on the other - it's changing of the original mechanics. We've already been offended for making "not pure original copy of the game" at some forums. 😅 
And I don't like making such tiny adjustable features list...

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> I'm really hesitating about that. On one hand, allowing this makes gameplay more convenient. But on the other - it's changing of the original mechanics.
> ...
> And I don't like making such tiny adjustable features list...

Should we do a survey every time we want to improve the use of the game because it changes the strict functioning of the original game? :)

So everything should be exactly as in the original game, bugs, flaws, inconsistencies, etc., and no quality of life improvements. :D
I agree that the recreation should be as close as possible to the original game.
But in the case of the hero being able to be dismissed when located in a castle, I consider that a quality of life improvement.
I'm not opposed to listening to a strong argument that it's not possible to do this to fit the original game perfectly.

Look at the game this way:
A hero who has no more movement points can be dismissed if he is positioned on the adventure map but not if he is positioned in a castle.
A hero surrounded by eight opposing heroes (his movement is blocked) can be dismissed if he is positioned on the adventure map but not if he is positioned in a castle and his movement is blocked by an opposing hero.
Now find a plausible explanation for this.
A question to ask about this.

> We've already been offended for making "not pure original copy of the game" at some forums. sweat_smile

Here is the question to ask these detractors: :D
Why not just play the original game then?

**Reply by Branikolog** [reply]
@LeHerosInconnu 

>Should we do a survey every time we want to improve the use of the game because it changes the strict functioning of the original game? :)

Probably yes. :D 

I agree, that fheroes2 will have a bunch of differences, compared to the OG. But each of this changes should be discussed, as we could miss some important aspects. At least the majority of members should approve the changes, before we merge them. :) So, that's the reason this discussion was made here, right? ;)

**Reply by LeHerosInconnu** [reply]
@Branikolog.

> @LeHerosInconnu
> 
> > Should we do a survey every time we want to improve the use of the game because it changes the strict functioning of the original game? :)
> 
> Probably yes. :D

So we need a poll section. :D

> I agree, that fheroes2 will have a bunch of differences, compared to the OG. But each of this changes should be discussed, as we could miss some important aspects. At least the majority of members should approve the changes, before we merge them. :) So, that's the reason this discussion was made here, right? ;)

We should know why this is not possible in the original game.
If it is due to a technical aspect, the game code, or for another reason.
And as a last resort, I will propose it as an optional enhancement in my mod. :D
